### PR TITLE
Make localization keys compatible with agglutinative and/or SOV type languages

### PR DESCRIPTION
**Pull request by luixxiul**
Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

This PR makes localization keys compatible with agglutinative and/or SOV (Subject Object Verb) type languages, clarifying context with variables, to improve localization experience on Weblate.

type: defect
element-web notes: none

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Make localization keys compatible with agglutinative and/or SOV type languages ([\#10159](https://github.com/matrix-org/matrix-react-sdk/pull/10159)). Contributed by @luixxiul.

### CODE REVIEW

File: src/i18n/strings/en_EN.json [hunk 1]
```diff
@@ -1574,8 +1574,8 @@
     "FAQ": "FAQ",
     "Keyboard Shortcuts": "Keyboard Shortcuts",
     "Versions": "Versions",
-    "Homeserver is": "Homeserver is",
```
Comment (luixxiul) [inline src/i18n/strings/en_EN.json:1577]: The position of verbs should not be hard coded.

File: src/components/views/dialogs/devtools/RoomNotifications.tsx [hunk 2]
```diff
@@ -39,22 +39,56 @@ export default function RoomNotifications({ onBack }: IDevtoolsProps): JSX.Eleme
...
+                                      strong: (sub) => <strong>{sub}</strong>,
+                                  },
+                              )
+                            : _t(
+                                  "Room is <strong>not encrypted 🚨</strong>",
+                                  {},
+                                  {
+                                      strong: (sub) => <strong>{sub}</strong>,
+                                  },
+                              )}
```
Comment (justjanne) [inline src/components/views/dialogs/devtools/RoomNotifications.tsx:91]: Can't you move the ternary inside the _t, so you can reuse the substitution?

Comment (justjanne) [reply src/components/views/dialogs/devtools/RoomNotifications.tsx:91]:
```ts
{_t(cli.isRoomEncrypted(room.roomId!)
    ? "Room is <strong>encrypted ✅</strong>"
    : "Room is <strong>not encrypted 🚨</strong>",
    {},
    {
        strong: (sub) => <strong>{sub}</strong>,
    },
)}
```

File: src/components/views/dialogs/devtools/RoomNotifications.tsx [hunk 3]
```diff
@@ -39,22 +39,56 @@ export default function RoomNotifications({ onBack }: IDevtoolsProps): JSX.Eleme
...
-                                {_t(", count:")} <strong>{count}</strong>
+                                {_t(
+                                    ", count: <strong>%(roomUnreadCount)s</strong>",
+                                    {
+                                        roomUnreadCount: count,
+                                    },
+                                    {
+                                        strong: (sub) => <strong>{sub}</strong>,
+                                    },
+                                )}
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/dialogs/devtools/RoomNotifications.tsx:61]: This should be done by count alone. Pluralisation can alternate translations and thus you can specify `|zero` to not have the `count: ` bit

[the remark this answers is not in the document]
Comment (justjanne) [reply src/components/views/dialogs/devtools/RoomNotifications.tsx:61]: That'd be even better then!

File: src/components/views/dialogs/devtools/RoomNotifications.tsx (hunk 2 above)
Comment (t3chguy) [reply src/components/views/dialogs/devtools/RoomNotifications.tsx:91]:
```
{_t(cli.isRoomEncrypted(room.roomId!)
    ? _td("Room is <strong>encrypted ✅</strong>")
    : _td("Room is <strong>not encrypted 🚨</strong>"),
    {},
    {
        strong: (sub) => <strong>{sub}</strong>,
    },
)}
```

*should work* (untested and other terms and conditions you can imagine)

File: src/components/views/dialogs/devtools/RoomNotifications.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (luixxiul) [reply src/components/views/dialogs/devtools/RoomNotifications.tsx:61]: Would you mind where to check to implement pluralization?

[the remark this answers is not in the document]
Comment (justjanne) [reply src/components/views/dialogs/devtools/RoomNotifications.tsx:61]: To use pluralization for this, you'd change the code in RoomNotifications.tsx to 

```ts
                    <li>
                        {_t(
                            "Room unread status: <strong>%(status)s</strong>, count: <strong>%(count)s</strong>",
                            {
                                status: humanReadableNotificationColor(color),
                                count,
                            },
                            {
                                strong: (sub) => <strong>{sub}</strong>,
                            },
                        )}
                    </li>
```

and then in the translations you'd add a variant with a pluralization of zero (the one with pluralization `other` is auto-generated by `yarn i18n`):

```json
"Room unread status: <strong>%(status)s</strong>, count: <strong>%(count)s</strong>|zero": "Room unread status: <strong>%(status)s</strong>",
"Room unread status: <strong>%(status)s</strong>, count: <strong>%(count)s</strong>|other": "Room unread status: <strong>%(status)s</strong>, count: <strong>%(count)s</strong>",
```

[the remark this answers is not in the document]
Comment (luixxiul) [reply src/components/views/dialogs/devtools/RoomNotifications.tsx]: I tried above, but it looks to create only the `|other` line for me, and `|zero` line is not generated. I am wondering how to let `yarn run i18n` generate the `|zero` line.

[the remark this answers is not in the document]
Comment (justjanne) [reply src/components/views/dialogs/devtools/RoomNotifications.tsx]: You'll have to add it manually

[the remark this answers is not in the document]
Comment (luixxiul) [reply src/components/views/dialogs/devtools/RoomNotifications.tsx]: OK I see, working on that. Thanks!

[the remark this answers is not in the document]
Comment (luixxiul) [reply src/components/views/dialogs/devtools/RoomNotifications.tsx]: Added with https://github.com/matrix-org/matrix-react-sdk/pull/10159/commits/e62af0577cbe8e26f480b310a162ec554a4e1c0d.

**Comment by luixxiul** [comment]
(https://github.com/matrix-org/matrix-react-sdk/pull/10159#event-8566701090 I didn't notice that asking a review from more than one reviewer was not possible)

**Review by justjanne — APPROVED** [review]

**Pull request merged**